src/status_im/transport/message/core.cljs

```diff
@@ -45,31 +34,34 @@
         messages (.-messages response-js)]
     (cond
       (seq installations)
-      (let [installation (.pop installations)]
+      (let [installations-clj (types/js->clj installations)]
+        (js-delete response-js "installations")
         (fx/merge cofx
                   {:utils/dispatch-later [{:ms 20 :dispatch [::process response-js]}]}
-                  (models.pairing/handle-installation (types/js->clj installation))))
+                  (models.pairing/handle-installations installations-clj)))
 
       (seq contacts)
-      (let [contact (.pop contacts)]
+      (let [contacts-clj (types/js->clj contacts)]
+        (js-delete response-js "contacts")
         (fx/merge cofx
-                  ;;TODO temporary fix for release, we have and issue with contacts updates , UI is really slow
-                  ;;we need to inspect all subsctiptions and views, but for now to temporary make it better
-                  ;; we use dispatch instead dispatch-later
-                  {:dispatch [::process response-js]}
-                  (handle-contact (-> contact (types/js->clj) (data-store.contacts/<-rpc)))))
+                  {:utils/dispatch-later [{:ms 20 :dispatch [::process response-js]}]}
+                  (handle-contacts (map data-store.contacts/<-rpc contacts-clj))))
 
       (seq chats)
-      (let [chat (.pop chats)]
+      (let [chats-clj (types/js->clj chats)]
+        (js-delete response-js "chats")
         (fx/merge cofx
                   {:utils/dispatch-later [{:ms 20 :dispatch [::process response-js]}]}
-                  (handle-chat (-> chat (types/js->clj) (data-store.chats/<-rpc)))))
+                  (handle-chats (map #(-> %
+                                          (data-store.chats/<-rpc)
+                                          (dissoc :unviewed-messages-count))
+                                     chats-clj))))
 
       (seq messages)
-      (let [message (.pop messages)]
+      (let [messages-clj (types/js->clj (.splice messages 0 50))]
```

Comment:
I would probably have this `50` somewhere more visible, some constants in `config` or top of the file